importlib_metadata/__init__.py

```diff
@@ -1071,6 +1070,9 @@ def _topmost(name: PackagePath) -> Optional[str]:
     return top if rest else None
 
 
+inspect = None
```

Comment:
This is a trick that is used in stdlib. I am not sure it is needed here, but since `get_toplevel_name` since to be called in a loop perhaps it is warranted to avoid the overhead of calling `import inspect` repeatedly.

Reply:
I'm not a fan of global variables or the additional complexity. This change introduces enough disruption to the essential logical flow that I'm -1. Do we know how much overhead there is in repeated `import inspect`? My understanding (which may be incorrect) is that `import inspect` is essentially a dict lookup if it's already been imported. I'm guessing the overhead is acceptable. Can we try a simple deferral for now?